### PR TITLE
get python version

### PR DESCRIPTION
**Pull request by TheoMF**

### CODE REVIEW

File: gepetuto/test.py [hunk 1]
```diff
@@ -10,16 +10,30 @@
 
 def test(**kwargs):
     """Test python scripts."""
+    python_version = kwargs["p"]
+    if python_version == "":
+        python_version = retrieve_python_version()
```
Comment (nim65s) [inline gepetuto/test.py:15]: I think this should go in the `default` value of `argparse.add_argument`

File: gepetuto/test.py [hunk 2]
```diff
@@ -10,16 +10,30 @@
...
     LOG.info("test passed.")
+
+
+def retrieve_python_version():
+    """Retrieve installed python version."""
+    version = sys.version_info[0]
+    if version == 3:
+        return "python3"
+    if version == 2:
+        return "python"
```
Comment (nim65s) [inline gepetuto/test.py:37]: I don't think we want to be compatible with python 2, but python 3 is sometimes called `python3`, sometimes called `python`, depending on the system

Comment (TheoMF) [reply gepetuto/test.py:37]: if we have python 3, it's sometimes possible to call it with python, but is it always possible to call it with `python3` ?

Comment (nim65s) [reply gepetuto/test.py:37]: no, some systems have the `python3` command but no `python`, while other systems have the `python` command but no `python3`.

**Comment by TheoMF** [comment]
j'ai repush

File: gepetuto/__main__.py [hunk 3]
```diff
@@ -67,6 +74,19 @@ def get_tp_id():
         current_tp_id += 1
 
 
+def retrieve_python_version():
+    """Retrieve installed python version."""
+    for path in sys.path:
```
Comment (nim65s) [inline gepetuto/__main__.py:79]: `sys.path` ? This is a list of directories where we should have python modules. The intepreter is provided at `sys.executable`.

If your goal is to find if there is an executable `python` or `python3`, I guess we can simply use a try/catch with a check_call on `python --version`: if it raise a `FileNotFoundError`, this is not a valid interpreter.

Comment (TheoMF) [reply gepetuto/__main__.py:79]: yes I was looking for the executable, but if we use `python --version`,  won't we will still not know if we should use `python` or `python3` ?

File: gepetuto/__main__.py [hunk 4]
```diff
@@ -67,6 +75,20 @@ def get_tp_id():
...
+    """Retrieve installed python version."""
+    try:
+        check_call(["python3", "--version"])
+        return "python3"
+    except FileNotFoundError:
+        try:
+            check_call(["python", "--version"])
+            return "python"
+        except FileNotFoundError:
+            LOG.warn("Didn't found python executable, using ", sys.executable)
```
Comment (nim65s) [inline gepetuto/__main__.py:88]:
```suggestion
            LOG.warn("Didn't found 'python3' or 'python' executable, using ", sys.executable)
```

**Review by nim65s — APPROVED** [review]
Ok, now this looks good, thanks !

Maybe we can just rename `python_version` into `python_interpreter`, or `python_executable` in the variable name, arguments and the retreive function, as this is more clear ? Or even just `python`, as you prefer.

**Pull request merged**